src/Native/System.Native/pal_networking.cpp

```diff
@@ -2376,7 +2376,7 @@ static void ConvertEventEPollToSocketAsync(SocketEvent* sae, epoll_event* epoll)
     uint32_t events = epoll->events;
     if ((events & EPOLLHUP) != 0)
     {
-        events = (events & ~EPOLLHUP) | EPOLLIN | EPOLLOUT;
+        events = (events & UnsignedCast(~EPOLLHUP)) | EPOLLIN | EPOLLOUT;
```

Comment:
I will change it to `static_cast<uint32_t>`, which I originally had, as
`(events & static_cast<uint32_t>(~EPOLLHUP)) == (events & ~EPOLLHUP)`